scene/resources/gradient.cpp

```diff
@@ -144,14 +144,16 @@ void Gradient::set_offset(int pos, const float offset) {
 	ERR_FAIL_COND(pos < 0);
 	if (points.size() <= pos) {
 		points.resize(pos + 1);
+		is_sorted = false;
 	}
+	_update_sorting();
```

Comment:
It seems the update should be after `points.write[pos].offset = offset;`, otherwise the offset might be set on the wrong point.

Reply:
I think calling `_update_sorting()` before setting the point and setting `is_sorted=false` after is the right way to handle this.